app/modules/code_provider/code_provider_service.py

```
@@ -1,26 +1,52 @@
     def get_repo(self, repo_name):
-        return self.service_instance.get_repo(repo_name)
+        # If it looks like a file path, use local service
+        if os.path.exists(repo_name):
+            return self.local_service.get_repo(repo_name)
+        # Otherwise, use GitHub service if configured
+        elif self.github_service:
+            return self.github_service.get_repo(repo_name)
+        else:
+            raise Exception(
+                f"GitHub not configured and {repo_name} is not a local path"
+            )
```

Ruff (0.14.3)

Create your own exception

(TRY002)